app/controllers/api/v1/base_controller.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class BaseController < ActionController::API
+      include CanCan::ControllerAdditions
+      include RequestTimeouts
+      include Pagy::Backend
+      include JsonApiPagination
+      include RaisingParameters
+
+      check_authorization
+
+      attr_accessor :current_api_user
+
+      before_action :authenticate_user
+
+      rescue_from StandardError, with: :error_during_processing
+      rescue_from CanCan::AccessDenied, with: :unauthorized
+      rescue_from ActiveRecord::RecordNotFound, with: :not_found
+      rescue_from Pagy::VariableError, with: :invalid_pagination
+      rescue_from ActionController::ParameterMissing, with: :missing_parameter
+      rescue_from ActionController::UnpermittedParameters, with: :unpermitted_parameters
+
+      private
+
+      def authenticate_user
+        return if (@current_api_user = request.env['warden'].user)
+
+        if api_key.blank?
+          # An anonymous user
+          @current_api_user = Spree::User.new
+          return
+        end
+
+        return if (@current_api_user = Spree::User.find_by(spree_api_key: api_key.to_s))
+
+        invalid_api_key
+      end
+
+      def current_ability
+        Spree::Ability.new(current_api_user)
+      end
+
+      def api_key
+        request.headers["X-Api-Token"] || params[:token]
+      end
+
+      def error_during_processing(exception)
+        Bugsnag.notify(exception)
+
+        if Rails.env.development? || Rails.env.test?
+          render status: :unprocessable_entity,
+                 json: json_api_error(exception.message, meta: exception.backtrace)
+        else
+          render status: :unprocessable_entity,
+                 json: json_api_error(I18n.t(:unknown_error, scope: "api"))
+        end
+      end
```

Comment:
We respond with unprocessable entity when encountering a StandardError. I reckon that the StandardError case should only happen on unexpected errors in which case a 500 server error may be the right response here. Opinions?

Reply:
I personally don't like the `unprocessable entity` error, which says nothing at all (and make me thing of an error due to client mistake).